src/components/textarea/bl-textarea.ts

```diff
@@ -0,0 +1,235 @@
+  labelFixed = false;
+
+  /**
+   * Sets placeholder of the textarea
+   */
+  @property({})
+  placeholder?: string;
+
+  /**
+   * Character counter
+   */
+  @property({ type: Boolean, attribute: 'character-counter' })
+  characterCounter = false;
+
+  /**
+   * Adds help text
+   */
+  @property({ type: String, attribute: 'help-text' })
+  helpText?: string;
+
+  /**
+   * Sets minimum length of the textarea
+   */
+  @property({ type: Number })
+  minlength?: number;
+
+  /**
+   * Sets max length of textarea
+   */
+  @property({ type: Number })
+  maxlength?:number;
+
+  /**
+   * Sets initial value of the textarea
+   */
+  @property()
+  value = '';
+
+  /**
+   * Sets input size.
+   */
+  @property({ type: Number})
+  rows?: number = 4;
+
+  @event('bl-input') private onInput: EventDispatcher<string>;
+
+  @event('bl-change') private onChange: EventDispatcher<string>;
+
+  @event('bl-invalid') private onInvalid: EventDispatcher<ValidityState>;
+
+
+  private lineHeight:number;
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.addEventListener('keydown', this.onKeydown);
+    this.addEventListener('invalid', this.onError);
+
+    if(this.expand){
+      this.addEventListener('input', this.autoResize, false);
+    }
+
+    this.internals.form?.addEventListener('submit', () => {
+      this.reportValidity();
+    });
+  }
+
+  private onKeydown = (event: KeyboardEvent): void => {
+    if (event.code === 'Enter' && this.form) {
+      submit(this.form);
+    }
+  }
+
+  private onError = (): void => {
+    this.onInvalid(this.internals.validity);
+  }
+
+  private inputHandler(event: Event) {
+    const value = (event.target as HTMLInputElement).value;
+
+    this.setValue(value);
+    this.onInput(value);
+  }
+
+  private changeHandler(event: Event) {
+    const value = (event.target as HTMLInputElement).value;
+
+    this.dirty = true;
+    this.setValue(value);
+    this.onChange(value);
+  }
+
+  reportValidity() {
+    this.dirty = true;
+    return this.checkValidity();
+  }
+
+  valueChangedCallback(value: string): void {
+    this.value = value;
+  }
+
+  validityCallback(): string | void {
+    return this.validationTarget?.validationMessage;
+  }
+
+  autoResize() {
+    const scrollHeight = this.validationTarget.scrollHeight;
+    this.validationTarget.style.height = "";
+    this.validationTarget.style.height = scrollHeight + "px";
+    console.log(scrollHeight, this.lineHeight);
+    this.rows = scrollHeight / this.lineHeight;
+  }
+
+  @state() private dirty = false;
+
+
+  render(): TemplateResult{
+    const maxLengthInvalid = this.internals.validity.rangeOverflow;
+    const invalidMessage = !this.checkValidity()
+      ? html`<p class="invalid-text">${this.validationMessage}</p>`
+      : ``;
+    const helpMessage = this.helpText ? html`<p class="help-text">${this.helpText}</p>` : ``;
+
+    const label = this.label ? html`<label>${this.label}</label>` : '';
```

Comment:
Give an `id` to textarea and put label `for` attribute.

Reply:
I added id that named "bl-text-area". But maybe you have better suggestion abi 😄

Reply:
Since this is inside ShadowDOM and there is not risk to have a conflict with other elements outside, you can use whatever you want. You don’t need to use any prefix.